dht.go

```diff
@@ -710,9 +750,9 @@ func (dht *IpfsDHT) handleProtocolChanges(proc goprocess.Process) {
 				// TODO: discuss how to handle this case
```

Comment:
should this also trigger an update?

Reply:
@willscott Great spot ! It will after [this code](https://github.com/libp2p/go-libp2p-kad-dht/blob/c649402191b967192873b158f5950734110c43cc/subscriber_notifee.go#L162) in Adin's PR goes in.